components/brave_shields/browser/ad_block_service_helper.cc

```diff
@@ -12,11 +12,29 @@
 #include "base/logging.h"
 #include "base/strings/string_util.h"
 #include "base/values.h"
+#include "net/base/registry_controlled_domains/registry_controlled_domain.h"
 
 using adblock::FilterList;
 
 namespace brave_shields {
 
+// Extracts the start and end characters of a domain from a hostname.
+// Required for correct functionality of adblock-rust.
+void domainResolverImpl(const char* host, uintptr_t* start, uintptr_t* end) {
```

Comment:
Maybe this could be named something like `AdBlockServiceDomainResolver` for clarity. Would also give it an initial cap if possible, to follow the Chromium coding guidelines.

Reply:
updated :+1: